middle_end/flambda2/numbers/numeric_types.ml

```diff
@@ -250,3 +250,59 @@ module Int64 = struct
 
   let cross_product = Pair.create_from_cross_product
 end
+
+module type Vector_width = sig
+  val size_in_int64s : int
+end
+
+module Vector_by_bit_pattern (Width : Vector_width) = struct
+  module T0 = struct
+    type t = Int64.t Array.t
+
+    let rec compare l r i =
+      if i = Width.size_in_int64s
+      then 0
+      else
+        let cmp = Int64.compare l.(i) r.(i) in
+        if cmp = 0 then compare l r (i + 1) else cmp
+
+    let compare l r = compare l r 0
+
+    let equal = Array.for_all2 Int64.equal
+
+    let hash v = Hashtbl.hash v
+
+    let print ppf t =
+      Format.pp_print_list
+        ~pp_sep:(fun ppf () -> Format.pp_print_char ppf ':')
+        Int64.print ppf (Array.to_list t)
+  end
+
+  include T0
+  module Self = Container_types.Make (T0)
+  include Self
+
+  let zero = Array.init Width.size_in_int64s (fun _ -> 0L)
+
+  let to_int64_array t = t
+
+  let of_int64_array t = t
```

Comment:
I think this should include a length check. Or use an explicit name like `of_int64_array_unsafe`.

Reply:
Added